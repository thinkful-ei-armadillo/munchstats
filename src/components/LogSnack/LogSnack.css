
.addIngredientMeasureForm {
  width:60%;
  margin: 30px auto;
}


#mealSelect, #mealTag{
  width: 200px;
}

.form-control{
  width: 200px;
  margin: 5px 0 5px 0;
  height: 25px;
  padding-left: 5px;
}


.rdtPicker{
  display: flex;
  justify-content: center;
  align-items: center;
}

.rdtPrev, .rdtSwitch, .rdtNext, .rdtDay{
  cursor: pointer;
}

.rdtDays tfoot tr td{
  text-align: center;
  padding: 10px;
  border: 1px black solid;
  cursor: pointer;
}
