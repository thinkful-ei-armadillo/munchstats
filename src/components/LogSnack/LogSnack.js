import React, { Component } from 'react';
import './App.css';

export default class LogSnack extends Component {

  getMealsByDate(start, end){

    fetch(`${config.API_ENDPOINT}/events/date`, {
      method: 'POST',
      body: JSON.stringify({
        'start': start,
        'end': end
      }),
      headers: {
        'content-Type': 'application/json',
      }
    })
      .then(res => res.json())
      .then(results => {
        const sortedEvents = this.sortEvents(results);
        // either set state or context here
        this.setState({ userEvents: sortedEvents });
      })
      .catch(err => console.log(err));
  }
  
  
  
  sortEvents = events => {
    return events.sort(function(a,b){
      if (a.date < b.date) {
        return 1;
      }
      if (a.date > b.date) {
        return -1;
      }
      return 0;
    });
  }
  
  
  // NOTE: START AND END FORMAT SHOULD BE: 2011-12-25 19:00:00 AS STRINGS

  render() {
    return (
      <div>
        
      </div>
    );
  }
}