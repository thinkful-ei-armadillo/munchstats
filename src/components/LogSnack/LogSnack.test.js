import React from 'react';
import LogSnack from './LogSnack';
import {MemoryRouter} from 'react-router-dom';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

describe('<LogSnack />', () => {
  it('Renders without crashing', () => {
    mount(<MemoryRouter><LogSnack /></MemoryRouter>);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><LogSnack /></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});