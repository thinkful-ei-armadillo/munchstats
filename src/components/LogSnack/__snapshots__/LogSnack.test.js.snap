// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogSnack /> renders the UI as expected 1`] = `
<div
  className="mealLogContainer"
>
  <h3>
    Log a Single Item Snack!
  </h3>
  <form
    className="mealForm"
    onSubmit={[Function]}
  >
    <label
      htmlFor="ingredient-input"
    >
      Ingredient
    </label>
    <input
      id="ingredient-input"
      name="ingredient-input"
      onChange={[Function]}
      required={true}
      value=""
    />
    <button
      className="Button"
      type="submit"
    >
      Search ingredients
    </button>
  </form>
  <section
    className="results"
  />
</div>
`;
